vignettes/BiocParallel_BatchtoolsParam.Rmd

```diff
@@ -0,0 +1,229 @@
+       well. If nothing is given, a temporary directory will be used
+       in your current working directory.
+        
+     + `work.dir` Working directory for R process for
+       running jobs.
+       
+     + `packages` Packages that will be loaded on each node.
+
+     + `namespaces` Namespaces that will be loaded on each
+        node.
+
+     + 'source` Files that are sourced before executing a
+        job.
+
+     + `load` Files that are loaded before executing a job.
+     
+    ```{r}
+        registryargs <- batchtoolsRegistryargs(
+        file.dir = "mytempreg",
+        work.dir = getwd(),
+        packages = character(0L),
+        namespaces = character(0L),
+        source = character(0L),
+        load = character(0L)
+        )
+        param <- BatchtoolsParam(registryargs = registryargs)
+        param
+    ```
+
+    + `resources` A named list of key-value pairs to be subsituted into the template
+    file; see `?batchtools::submitJobs`.
+
+     + `template` The template argument is unique to the class `BatchtoolsParam`. It is required by the
+    job scheduler. It defines how the jobs are submitted to the job
+    scheduler. If the template is not given and the cluster is chosen, a
+    default template is selected from the batchtools package.
+
+     + `log` The log option is logical, TRUE/FALSE. If it is set to TRUE, then
+    the logs which are in the registry are copied to directory given by
+    the user using the `logdir` argument.
+
+    + `logdir` Path to the logs. It is given only if `log=TRUE`.
+
+    + `resultdir` Path to the directory is given when the job has files to be saved in
+    a directory.
+
+# Defining templates
+
+The job submission template controls how the job is processed by the job
+scheduler on the cluster. Obviously, the format of the template will
+differ depending on the type of job scheduler. Let's look at the default
+SLURM template as an example:
+
+```{r}
+fname <- batchtoolsTemplate("slurm")
+cat(readLines(fname), sep="\n")
+```
+
+The `<%= =>` blocks are automatically replaced by the values of the elements in
+the `resources` argument in the`BatchtoolsParam` constructor. Failing to specify critical parameters
+properly (e.g., wall time or memory limits too low) will cause jobs to
+crash, usually rather cryptically. We suggest setting parameters
+explicitly to provide robustness to changes to system defaults. Note
+that `<%= =>` the blocks themselves do not usually need to be modified in the
+template.
+
+The part of the template that is most likely to require explicit
+customization is the last line containing the call to `Rscript`. A more
+customized call may be necessary if the R installation is not standard,
+e.g., if multiple versions of R have been installed on a cluster. For
+example, one might use instead:
+
+````{verbatim}
+    echo 'batchtools::doJobCollection("<%= uri %>")' |\
+        ArbitraryRcommand --no-save --no-echo
+````
```

Comment:
I think when something is "verbatim", it's code that shouldn't be executed. This looks like bash code. Replace with

    ```{bash doJobCollection, eval=FALSE}
        echo 'batchtools::doJobCollection("<%= uri %>")' |\
            ArbitraryRcommand --no-save --no-echo
    ```